.github/workflows/autorelease.yml

```diff
@@ -0,0 +1,105 @@
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Only run on merged PRs with an appropriate label.
+    if: |
+      github.event.pull_request.merged &&
+      (
+        contains(github.event.pull_request.labels.*.name, 'patch') ||
+        contains(github.event.pull_request.labels.*.name, 'minor') ||
+        contains(github.event.pull_request.labels.*.name, 'major') ||
+        contains(github.event.pull_request.labels.*.name, 'alpha') ||
+        contains(github.event.pull_request.labels.*.name, 'beta') ||
+        contains(github.event.pull_request.labels.*.name, 'rc')
+      )
+    
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.base_ref }}
+          fetch-depth: 0 
+
+      - name: Set Git user
+        run: |
+          git config --global user.email "action@github.com"
+          git config --global user.name "Github Action"
+
+      # Run 'npm run package' and commit dist folder if there are any changes.
+      - name: npm run package
+        run: |
+          NEW_TAG=$(git tag --sort=-taggerdate | head -1)
+          npm run package
+          if [ $(git status | grep "dist\/") ]
+          then 
+            git add dist
+            git commit -m "Updated build package for release $NEW_TAG"
+          fi
+      
+      # Increment version based on label
+      - name: Patch Release tag
+        if: ${{ contains( github.event.pull_request.labels.*.name, 'patch') }}
+        run: npm version patch
+
+      - name: Minor Release tag
+        if: ${{ contains( github.event.pull_request.labels.*.name, 'minor') }}
+        run: npm version minor
+      
+      - name: Major Release tag
+        if: ${{ contains( github.event.pull_request.labels.*.name, 'major') }}
+        run: npm version major
+      
+      - name: Alpha Release tag
+        if: ${{ contains( github.event.pull_request.labels.*.name, 'alpha') }}
+        run: npm version prerelease --preid alpha
+      
+      - name: Beta Release tag
+        if: ${{ contains( github.event.pull_request.labels.*.name, 'beta') }}
+        run: npm version prerelease --preid beta
+
+      - name: RC Release tag
+        if: ${{ contains( github.event.pull_request.labels.*.name, 'rc') }}
+        run: npm version prerelease --preid rc
+
+      # Automatically append new commit messages to changelog.
+      - name: "Update changelog"
+        run: |
+          NEW_TAG=$(git tag --sort=-taggerdate | head -1)
+          PREVIOUS_TAG=$(git tag --sort=-taggerdate | sed -n '2 p')
+          echo -e "$(git log --no-merges --pretty=format:%s --invert-grep --grep="Updated changelog with notes for" $PREVIOUS_TAG..$NEW_TAG)\n" >> CHANGELOG.txt
+          git add CHANGELOG.txt
+          git commit -m "Updated changelog with notes for $NEW_TAG"
```

Comment:
Any reason to not use something like https://www.npmjs.com/package/auto-changelog?

Reply:
@sherakama This wasn't on my radar, but I'm not opposed to giving it a try.

Reply:
Give it a go then. It has nice formatting.

Reply:
@sherakama Action has been refactored to use auto changelog script.